components/server/src/ome/services/sharing/ShareBean.java

```diff
@@ -752,6 +760,35 @@ public void invalidateConnection(long shareId, Experimenter exp) {
         throw new UnsupportedOperationException();
     }
 
+    @RolesAllowed("user")
+    @Transactional(readOnly = false)
+    public void notifyMembersOfShare(long shareId, String subject, String message,
+            boolean html, List<Experimenter> exps) {
+
+        Set<Long> memberIds = new HashSet<Long>();
+        for (final Experimenter e : getAllMembers(shareId)) {
+            memberIds.add(e.getId());
+        }
+
+        Map<Experimenter, String> errors = new HashMap<Experimenter, String>();
+        for (final Experimenter e : exps) {
+            if (memberIds.contains(e.getId())){
+                if (e.getEmail() != null && mailUtil.validateEmail(e.getEmail())) {
+                    try {
+                        mailUtil.sendEmail(e.getEmail(), subject, message, html,
+                                null, null);
+                    } catch (MailException me) {
+                        errors.put(e, me.getMessage());
+                    }
+                }
+            }
+        }
+        if (!errors.isEmpty()) {
+            ServiceHandler sh = new ServiceHandler(new CurrentDetails());
```

Comment:
Ideally if getResultsString should be is in utile or somewhere. That is very useful for logging a long list.